### PR TITLE
Add ability to clear DNS entries on static DNS

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds the ability to clear the list of DNS entries in Windows by passing an empty list.

Execution Module:
- Passing no DNS servers will clear the list because it is using *args.
- Update documentation
- Added a note about clearing DNS entries

State Module
- An empty list in `dns_servers` will clear the entires
- `None` in `dns_servers` will do nothing (default behavior)
- Update Documentation

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/49730

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/win_ip.py [hunk 1]
```diff
@@ -319,13 +319,35 @@ def set_static_dns(iface, *addrs):
...
     Set static DNS configuration on a Windows NIC
 
+    Args:
+
+        iface (str): The name of the interface to set
+
+        addrs (*):
+            One or more DNS servers to be added
+
+            .. note::
```
Comment (cachedout) [inline salt/modules/win_ip.py]: This does not seem like the right behavior to me. These sorts of side-effects are implicit and, in my opinion, should be explicit. IMHO, nothing passed should mean no change is applied and a list should only be cleared if, and only if, an empty list is explicitly passed in.

**Comment by twangboy** [comment]
@cachedout Only clears the list if you specifically pass []
@dwoz Added some tests

**Review by dwoz — APPROVED** [review]

**Comment by twangboy** [comment]
I don't think all the tests actually ran... 4 seems like a small number

**Comment by rallytime** [comment]
@twangboy Yes, 4 is not the right number. The rest of the tests are waiting in the jenkins queue for an executor slot. Once they start, this will update with the remaining jobs.

**Comment by rallytime** [comment]
@twangboy Your changes have caused a test to fail: https://jenkinsci.saltstack.com/job/pr-kitchen-ubuntu1604-py2/job/PR-49763/4/

**Pull request merged**